samples/bluetooth/central_otc/src/main.c

```diff
@@ -0,0 +1,585 @@
+	gpio_add_callback(btn.port, &button_cb_data);
+
+	printk("Set up button at %s pin %d\n", btn.port->name, btn.pin);
+}
+
+static void configure_buttons(void)
+{
+	gpio_init_callback(&button_cb_data, button_pressed, 0);
+
+	for (uint8_t idx = 0; idx < BTN_COUNT; idx++) {
+		configure_button_irq(btns[idx]);
+	}
+}
+
+static bool eir_found(struct bt_data *data, void *user_data)
+{
+	bt_addr_le_t *addr = user_data;
+	int i;
+
+	switch (data->type) {
+	case BT_DATA_UUID16_SOME:
+	case BT_DATA_UUID16_ALL:
+		if (data->data_len % sizeof(uint16_t) != 0U) {
+			printk("AD malformed\n");
+			return true;
+		}
+
+		for (i = 0; i < data->data_len; i += sizeof(uint16_t)) {
+			struct bt_le_conn_param *param;
+			struct bt_uuid *uuid;
+			uint16_t u16;
+			int err;
+
+			memcpy(&u16, &data->data[i], sizeof(u16));
+			uuid = BT_UUID_DECLARE_16(sys_le16_to_cpu(u16));
+			if (bt_uuid_cmp(uuid, BT_UUID_OTS)) {
+				continue;
+			}
+
+			err = bt_le_scan_stop();
+			if (err) {
+				printk("Stop LE scan failed (err %d)\n", err);
+				continue;
+			}
+
+			param = BT_LE_CONN_PARAM_DEFAULT;
+			err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, param, &default_conn);
+			if (err) {
+				printk("Create conn failed (err %d)\n", err);
+				start_scan();
+			}
+
+			return false;
+		}
+	}
+	return true;
+}
+
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	char dev[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(addr, dev, sizeof(dev));
+
+	/* We're only interested in connectable events and scan response
+	 * because service UUID is in sd of sample peripheral_ots.
+	 */
+	if (type == BT_GAP_ADV_TYPE_ADV_IND || type == BT_GAP_ADV_TYPE_ADV_DIRECT_IND ||
+	    type == BT_GAP_ADV_TYPE_SCAN_RSP) {
+		bt_data_parse(ad, eir_found, (void *)addr);
+	}
+}
+
+static void start_scan(void)
+{
+	int err;
+
+	/* Use active scanning and disable duplicate filtering to handle any
+	 * devices that might update their advertising data at runtime.
+	 */
+	struct bt_le_scan_param scan_param = {
+		.type = BT_LE_SCAN_TYPE_ACTIVE,
+		.options = BT_LE_SCAN_OPT_NONE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window = BT_GAP_SCAN_FAST_WINDOW,
+	};
+
+	err = bt_le_scan_start(&scan_param, device_found);
+	if (err) {
+		printk("Scanning OTS TAG failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning successfully started\n");
+}
+
+static void subscribe_func(void)
+{
+	printk("subscribe OACP and OLCP Indication\n");
+	oacp_sub_params = &otc.oacp_sub_params;
+	oacp_sub_params->disc_params = &otc.oacp_sub_disc_params;
+	if (oacp_sub_params) {
+		/* With ccc_handle == 0 it will use auto discovery */
+		oacp_sub_params->ccc_handle = 0;
+		oacp_sub_params->end_handle = otc.end_handle;
+		oacp_sub_params->value = BT_GATT_CCC_INDICATE;
+		oacp_sub_params->value_handle = otc.oacp_handle;
+		oacp_sub_params->notify = bt_ots_client_indicate_handler;
+		bt_gatt_subscribe(default_conn, oacp_sub_params);
```

Comment:
What if this fails?

Reply:
Should be considered as discovery failed. Add bt_gatt_subscribe return value checking.